client/core/market.go

```diff
@@ -167,3 +167,99 @@ func (derivativeMarket DerivativeMarket) MarginFromExtendedChainFormat(chainValu
 func (derivativeMarket DerivativeMarket) NotionalFromExtendedChainFormat(chainValue sdkmath.LegacyDec) decimal.Decimal {
 	return common.RemoveExtraDecimals(derivativeMarket.NotionalFromChainFormat(chainValue), AdditionalChainFormatDecimals)
 }
+
+type BinaryOptionMarket struct {
+	Id                  string
+	Status              string
+	Ticker              string
+	OracleSymbol        string
+	OracleProvider      string
+	OracleType          string
+	OracleScaleFactor   uint32
+	ExpirationTimestamp int64
+	SettlementTimestamp int64
+	QuoteToken          Token
+	MakerFeeRate        decimal.Decimal
+	TakerFeeRate        decimal.Decimal
+	ServiceProviderFee  decimal.Decimal
+	MinPriceTickSize    decimal.Decimal
+	MinQuantityTickSize decimal.Decimal
+	MinNotional         decimal.Decimal
+	SettlementPrice     *decimal.Decimal
+}
+
+func (market BinaryOptionMarket) QuantityToChainFormat(humanReadableValue decimal.Decimal) sdkmath.LegacyDec {
```

Comment:
Because depending on whether you are translating price, quantity or notional the calculation is different, and this functions abstract the user from the complexity